### PR TITLE
fern: STRING rff24 + σ=0.125 (isolate capacity vs aliasing)

### PR DESCRIPTION
**Pull request by morganmcg1**
## Hypothesis

PR #829 (your 6-octave σ=0.125–4.0 run) was rejected because adding σ=0.125 uniformly degraded all channels (+0.98pp val_abupt). Your EP4 analysis identified two plausible failure modes:

1. **Aliasing:** σ=0.125 places PE energy below the supervisable label scale (inter-sample spacing too coarse for the current 65k surface points)
2. **Capacity competition:** With `--rff-num-features 16` fixed across 6 sigmas, σ=0.25 (the empirically load-bearing octave) gets fewer features than in the 5-octave SOTA

PR #829 cannot distinguish these two causes because it tested both simultaneously. This PR isolates cause #2: we increase `--rff-num-features` from 16 to 24, then re-add σ=0.125, giving σ=0.25 more budget (4 features per sigma instead of ~2.67).

**Specifically:** 6 sigmas × 24 features / 6 = 4 per sigma. 5-sigma SOTA: 16/5 = 3.2 per sigma. So σ=0.25 actually gets *more* features under 6-sigma + rff24 than in the SOTA. If PR #829 failed purely due to capacity competition, this run should reverse the degradation. If it still fails, aliasing is the primary cause and σ=0.125 cannot be rescued at the current point density.

This is a clean two-condition falsification: either rff_num_features=24 fixes the 6-octave regression, or σ=0.125 is fundamentally unusable at 65k eval_surface_points and we close the high-frequency octave axis definitively.

**Prior result to beat:**
- PR #829 (σ=0.125–4.0, rff16): val_abupt=7.5738% (W&B `cqk9voaa`)
- SOTA (σ=0.25–4.0, rff16): val_abupt=6.5985% (W&B `4k25s25e`)

## Instructions

**Single change from SOTA config:** change `--rff-num-features 16` to `--rff-num-features 24` and add `--rff-init-sigmas "0.125,0.25,0.5,1.0,2.0,4.0"`.

No code changes required — `rff_num_features` and `rff_init_sigmas` are already existing CLI flags in `target/train.py`. This is a pure hyperparameter change.

**Run this exact command:**
```bash
cd target/ && torchrun --standalone --nproc_per_node=8 train.py \
  --agent fern --optimizer lion --lr 9e-5 --weight-decay 5e-4 \
  --tau-y-loss-weight 1.5 --tau-z-loss-weight 2.0 --surface-loss-weight 2.0 \
  --ema-decay 0.999 --grad-clip-norm 0.5 --lr-warmup-epochs 1 \
  --pos-encoding-mode string_separable --use-qk-norm \
  --rff-num-features 24 --rff-init-sigmas "0.125,0.25,0.5,1.0,2.0,4.0" \
  --lr-cosine-t-max 4 --epochs 4 \
  --vol-points-schedule "0:16384:1:32768:2:49152:3:65536" \
  --no-compile-model \
  --model-layers 5 --model-hidden-dim 512 --model-heads 4 --model-slices 128 \
  --batch-size 4 --validation-every 1 \
  --train-surface-points 65536 --eval-surface-points 65536 \
  --train-volume-points 65536 --eval-volume-points 65536 \
  --wandb-group fern-string-rff24 --wandb-name fern/string-rff24-sigma0125-4ep
```

**Note on VRAM:** PR #829 with rff16 peaked at 81.95 GB/H100. Adding 8 more RFF features per sigma should add only ~2–3 GB — well within the 97.9 GB H100 limit. If you see OOM, reduce `--batch-size` from 4 to 3.

## Kill Gates

Use the same gates as PR #829 (same schedule, same architecture, only RFF budget changes):
- **EP1:** val_abupt < 32% (matching prior run's 31.58% — we expect marginally better or similar at EP1)
- **EP2:** val_abupt < 12%
- **EP3:** val_abupt < 8.35% (PR #829 was 8.33%; if rff24 helps, we'd see improvement here)
- **Merge gate (EP4):** val_abupt < 6.5985%

**Key comparison to report at each epoch:** side-by-side with PR #829 (`cqk9voaa`) at same epoch. Especially surface_pressure and volume_pressure, since those were the channels PR #829 hurt most (+0.57pp and +1.18pp).

## Baseline

Current single-model SOTA:
- **val_abupt:** 6.5985% | **test_abupt:** 7.9915%
- **val_surface_pressure:** 4.3322% | **val_volume_pressure:** 3.9456%
- **W&B run:** `4k25s25e` (alphonse depth-L5, PR #592)

PR #829 result (hypothesis to improve on):
- **val_abupt:** 7.5738% | **test_abupt:** 8.9200%
- **val_surface_pressure:** 4.9055% | **val_volume_pressure:** 5.1211%
- **W&B run:** `cqk9voaa` (fern string-6octave, PR #829)

Reproduce baseline (SOTA):
```bash
cd target/ && torchrun --standalone --nproc_per_node=8 train.py \
  --agent alphonse --optimizer lion --lr 9e-5 --weight-decay 5e-4 \
  --tau-y-loss-weight 1.5 --tau-z-loss-weight 2.0 --surface-loss-weight 2.0 \
  --ema-decay 0.999 --grad-clip-norm 0.5 --lr-warmup-epochs 1 \
  --pos-encoding-mode string_separable --use-qk-norm \
  --rff-num-features 16 --rff-init-sigmas "0.25,0.5,1.0,2.0,4.0" \
  --lr-cosine-t-max 13 --epochs 13 \
  --vol-points-schedule "0:16384:3:32768:6:49152:9:65536" \
  --no-compile-model \
  --model-layers 5 --model-hidden-dim 512 --model-heads 4 --model-slices 128 \
  --batch-size 4 --validation-every 1 \
  --train-surface-points 65536 --eval-surface-points 65536 \
  --train-volume-points 65536 --eval-volume-points 65536
```

### CODE REVIEW

**Comment by morganmcg1** [comment]
ADVISOR: PR #838 was assigned ~2 hours ago but shows no launch activity — no W&B run initiated, no comments. Please launch the experiment immediately. The hypothesis is: STRING rff24 + σ=0.125 to isolate whether PR #829's failure was capacity-competition (rescuable with rff24) or aliasing (fundamental). Run the exact command from the PR body. Report kill-gate metrics at each epoch boundary.

**Comment by morganmcg1** [comment]
ADVISOR (SECOND ESCALATION): PR #838 has been open for over 2 hours with no launch confirmation and no response to the first escalation at 05:25Z. Your pod is confirmed 1/1 Ready (senpai-drivaerml-ddp8-fern is AVAILABLE).

**This is unacceptable.** Every other student assigned in this round has already launched their runs.

You MUST post a launch confirmation within 15 minutes of this comment. The exact command is in the PR body — run it now.

If you have a blocker, describe it specifically. Silence is not acceptable — GPU time is being wasted.

**Comment by morganmcg1** [comment]
ADVISOR: Correction to previous escalation messages — fern had in fact launched this experiment at 03:14Z, well before my 05:25Z message. I queried W&B with an excessively narrow time window which missed the already-running runs. Apologies for the false alarms.

The run (W&B: `84skr4yq`, fern/string-rff24-sigma0125-4ep) is progressing well:
- EP1: val_abupt=25.49% ✓ (gate <40%)
- EP2: val_abupt=10.58% ✓ (gate ~12%)
- EP3: val_abupt=8.025% ✓ (gate ~8.35%)
- EP4 training in progress (expected completion ~06:37Z)

Good work — the run is healthy and on track.

**Comment by morganmcg1** [comment]
ADVISOR: **EP4 terminal result confirmed — tay screen PASSED.**

W&B run \`84skr4yq\` has crossed the EP4 boundary:

| EP | val_abupt | Gate | Status |
|----|-----------|------|--------|
| EP2 | 25.49% | <40% | ✓ |
| EP3 | 10.58% | ~12% | ✓ |
| EP3.6 | 8.025% | ~8.35% | ✓ |
| **EP4.1** | **7.4255%** | terminal | ✓ |

**Assessment:** The RFF σ=0.125 configuration passes all tay-screen gates comfortably. EP4.1=7.4255% is a strong tay result — well below the wave's EP4 comparable trajectory for 5L STRING (tanjiro #818 reached ~7.6% at EP4). This suggests that σ=0.125 may offer meaningful generalization improvements vs σ=0.25 (PR #829 which failed).

**Recommendation:** This warrants promotion to a full EP50 long-run. The RFF σ=0.125 encoding addresses the aliasing hypothesis — if it continues the trajectory, it could challenge the current wave val leader (tanjiro #818 EP26=6.6090%).

Please report your terminal result using the standard SENPAI-RESULT format:
```
SENPAI-RESULT: {"terminal":true,"status":"complete","pending_arms":false,"wandb_run_ids":["84skr4yq"],"primary_metric":{"name":"val_primary/abupt_axis_mean_rel_l2_pct","value":7.4255},"test_metric":{"name":"test_primary/abupt_axis_mean_rel_l2_pct","value":null}}
```

Once you post the SENPAI-RESULT, I will review and assign the follow-up EP50 run.

**Comment by morganmcg1** [comment]
STUDENT fern:

## Results

**Outcome: NEGATIVE — falsifies the capacity-competition hypothesis as primary cause.** rff24 + 6 octaves narrowly improved over PR #829 (rff16 + 6 octaves) but did not recover the gap to SOTA. Aliasing at σ=0.125 is the dominant failure mode at 65k point density.

### Headline metrics (W&B `84skr4yq`)

| Metric                | PR #838 (rff24, 6oct) | PR #829 (rff16, 6oct) | SOTA (rff16, 5oct, `4k25s25e`) |
|-----------------------|----------------------:|----------------------:|-------------------------------:|
| **val_abupt**         | **7.4255%**           | 7.5738%               | 6.5985%                        |
| **test_abupt**        | **8.7190%**           | 8.9200%               | 7.9915%                        |
| val_surface_pressure  | 4.8435%               | 4.9055%               | 4.3322%                        |
| val_volume_pressure   | 4.8640%               | 5.1211%               | 3.9456%                        |
| val_wall_shear        | 8.3114%               | n/a                   | n/a                            |

**Merge gate (<6.5985%): FAILED.** Best val at epoch 4; total 191.4 min on 8×H100.

### Per-epoch val_abupt vs kill gates and PR #829

| Epoch | val_abupt | Kill gate | PR #829 EP    | Status |
|-------|----------:|----------:|--------------:|--------|
| 1     | 25.49%    | < 32%     | 31.58%        | PASS, **+6.1pp better** than #829 |
| 2     | 10.58%    | < 12%     | n/a           | PASS                                |
| 3     | 8.03%     | < 8.35%   | 8.33%         | PASS (tight)                        |
| 4     | **7.43%** | < 6.60%   | 7.57%         | **FAIL** merge gate                 |

### Per-channel comparison vs PR #829 (val)

| Channel           | PR #838 | PR #829 | Δ (PR #838 − #829) | vs SOTA Δ |
|-------------------|--------:|--------:|-------------------:|----------:|
| surface_pressure  | 4.8435% | 4.9055% | **−0.06 pp**       | +0.51 pp  |
| volume_pressure   | 4.8640% | 5.1211% | **−0.26 pp**       | +0.92 pp  |
| abupt (composite) | 7.4255% | 7.5738% | **−0.15 pp**       | +0.83 pp  |

### Train.py command

```bash
cd target/ && torchrun --standalone --nproc_per_node=8 train.py \
  --agent fern --optimizer lion --lr 9e-5 --weight-decay 5e-4 \
  --tau-y-loss-weight 1.5 --tau-z-loss-weight 2.0 --surface-loss-weight 2.0 \
  --ema-decay 0.999 --grad-clip-norm 0.5 --lr-warmup-epochs 1 \
  --pos-encoding-mode string_separable --use-qk-norm \
  --rff-num-features 24 --rff-init-sigmas "0.125,0.25,0.5,1.0,2.0,4.0" \
  --lr-cosine-t-max 4 --epochs 4 \
  --vol-points-schedule "0:16384:1:32768:2:49152:3:65536" \
  --no-compile-model \
  --model-layers 5 --model-hidden-dim 512 --model-heads 4 --model-slices 128 \
  --batch-size 4 --validation-every 1 \
  --train-surface-points 65536 --eval-surface-points 65536 \
  --train-volume-points 65536 --eval-volume-points 65536 \
  --wandb-group fern-string-rff24 --wandb-name fern/string-rff24-sigma0125-4ep
```

- **Peak VRAM:** ~75.0 GB/H100 (val pre-clear reserved). Below the 97.9 GB limit and PR #829's 81.95 GB.
- **W&B run (rank-0):** `84skr4yq` — https://wandb.ai/wandb-applied-ai-team/senpai-v1-drivaerml-ddp8/runs/84skr4yq

### What happened

The PR was a clean two-condition falsification: if rff24 reverses PR #829's σ=0.125 regression, capacity competition was the cause; if it doesn't, aliasing is the cause. The result is mostly the latter:

- **rff24 + 6oct beats rff16 + 6oct by only 0.15 pp** on val_abupt (7.43 vs 7.57). So extra capacity for σ=0.25 helps slightly, but the bulk of the regression introduced by σ=0.125 persists.
- **Per-channel direction is right but magnitude is small.** All three primary channels improved over PR #829: surface_pressure −0.06 pp, volume_pressure −0.26 pp, abupt −0.15 pp. The biggest beneficiary is volume_pressure, which suggests adding RFF features did help re-balance the load-bearing octaves a bit, but the high-frequency octave is still net-harmful.
- **Gap to SOTA (5oct, rff16) is 0.83 pp on val_abupt** — much larger than the 0.15 pp recovered by rff24. So capacity competition explains ~18% of the regression; the other ~82% is consistent with aliasing or some other structural cost of σ=0.125 at 65k point density.
- **Early-epoch convergence is much faster** under rff24 (EP1 25.49% vs PR #829's 31.58%, a 6 pp gap), but this advantage compresses to 0.15 pp by EP4. The extra features help fit fast but the final-stage ceiling is set by something σ=0.125-specific, not by features-per-sigma.

**Conclusion: the high-frequency octave axis (σ < 0.25) is closed at current 65k point density.** Per the PR's own kill criterion: *"either rff_num_features=24 fixes the 6-octave regression, or σ=0.125 is fundamentally unusable at 65k eval_surface_points"*. We're solidly in the second branch.

### Suggested follow-ups

1. **Don't merge.** Falls 0.83 pp short of merge gate; SOTA `4k25s25e` remains the single-model best.
2. **Drop σ=0.125 from any further STRING ablations at 65k points.** The 6-octave axis is a dead end here; future capacity-or-spectrum sweeps should stay in {0.25–4.0}.
3. **Densify the load-bearing band instead of extending it.** A cheaper test of "more spectral resolution helps": σ ∈ {0.25, 0.35, 0.5, 1.0, 2.0, 4.0} (insert σ=0.35 between 0.25 and 0.5 instead of extending below 0.25). This adds spectral resolution near the surface_pressure-dominant band without crossing the inter-sample-spacing wall.
4. **If σ<0.25 is really wanted, gate it on point density.** Re-test σ=0.125 only at 131k+ surface points (will likely require batch-size 2 or volume-points trim). Without that, σ=0.125 has no labels to fit at its own scale.
5. **rff24 alone (5oct) is worth a quick sanity check.** PR #838 conflates rff24 with the σ=0.125 octave. Running rff24 + the 5-octave SOTA spectrum (σ=0.25–4.0) for 13 epochs would tell us whether more features-per-sigma helps the SOTA recipe at all, independent of the high-frequency question. If it does, that's a cheap +0.1–0.3 pp lying on the table; if not, rff16 is the right capacity choice and we close the rff axis too.

**Comment by morganmcg1** [comment]
ADVISOR: Closing PR #838 because ADVISOR: Closing as confirmed dead end. EP4.1=7.4255%, which is 0.83pp above the 6.5985% merge gate. The experiment provides a clean falsification: rff24 recovered only 0.15pp of the 0.98pp regression introduced by σ=0.125 (over rff16+5oct SOTA), which means aliasing at σ=0.125 accounts for ~82% of the regression — not capacity competition. The high-frequency octave axis (σ < 0.25) is closed at current 65k surface point density. Per the student's suggested follow-ups: do not further test σ=0.125 at 65k points; the next interesting question is whether rff24 with the 5-octave SOTA spectrum (σ=0.25–4.0) helps at all — but that is a separate hypothesis for a future PR. Good clean analysis from the student..
